docs/reference/troubleshooting/troubleshooting-searches.asciidoc

```diff
@@ -0,0 +1,240 @@
+
+[[troubleshooting-searches]]
+== Troubleshooting searches
+
+When you query your data, Elasticsearch may return an error, no search results,
+or results in an unexpected order. This guide describes how to troubleshoot
+searches.
+
+[discrete]
+[[troubleshooting-searches-exists]]
+=== Ensure the data stream, index, or alias exists
+
+Elasticsearch returns an `index_not_found_exception` when the data stream, index
+or alias you try to query does not exist. This can happen when you misspell the
+name or when the data has been indexed to a different data stream or index.
+
+Use the <<indices-exists,exists API>> to check whether a data stream, index, or
+alias exists:
+
+[source,console]
+----
+HEAD my-data-stream
+----
+
+Use the <<data-stream-stats-api,data stream stats API>> to list all data
+streams:
+
+[source,console]
+----
+GET /_data_stream/_stats?human=true
+----
+
+Use the <<cat-indices,cat indices API>> to list all indices:
+
+[source,console]
+----
+GET _cat/indices?v=true&s=index&h=index,docs.count,status,health
+----
+
+Use the <<cat-alias,cat aliases API>> to list all aliases:
+
+[source,console]
+----
+GET _cat/aliases?v=true&s=alias&h=alias,index,is_write_index
+----
+
+[discrete]
+[[troubleshooting-searches-data]]
+=== Ensure the data stream or index contains data
+
+When a search request returns no hits, the data stream or index may contain no
+data. This can happen when there is a data ingestion issue. For example, the
+data may have been indexed to a data stream or index with another name.
+
+Use the <<search-count,count API>> to retrieve the number of documents in a data
+stream or index. Check that `count` in the response is not 0.
+
+////
+[source,console]
+----
+PUT my-index-000001
+----
+////
+
+[source,console]
+----
+GET /my-index-000001/_count
+----
+//TEST[continued]
+
+NOTE: When getting no search results in {kib}, check that you have selected the
+correct data view and a valid time range. Also, ensure the data view has been
+configured with the correct time field.
+
+[discrete]
+[[troubleshooting-searches-field-exists]]
+=== Ensure the field exists and has data
```

Comment:
I would give a more prominent role to the field_caps API: that can totally answer the "is the field there" question. I wonder if the exists query step is even needed then. I would consider showing the output for one field, for clarify.

Reply:
I think the exists query answers the "has data" question. I don't think field_caps answers that question? I think it makes sense to first check whether this is a field that returns any values, and if not, then move to field_caps as the next step?

Reply:
I would think the other way around. Field caps lets you know if the field is there and whether it is searchable or not. Then you may want to know if the field has values etc.

Reply:
I've pushed a change to reorder the steps so field caps comes earlier:

1. Check that the data stream, index or alias exists
2. Use _count to ensure the data stream, index or alias contain data
3. Use _field_caps to check that the field exists and its capabilities
4. Check mapping and text analysis
5. Check for values with the exists query and aggs
6. Validate, explain and profile
7. Check index settings